### ISSUE TITLE
Revert of nodejs/node#55623

### ISSUE THREAD

**Issue by Flarna** [nodejs/TSC#1659]
The PR https://github.com/nodejs/node/pull/55623 caused a new issue: https://github.com/nodejs/node/issues/56002.

https://github.com/nodejs/node/pull/55623 landed already in 23.2.0 and to my understanding of contributor docs [here](https://github.com/nodejs/node/blob/main/doc/contributing/collaborator-guide.md#unintended-breaking-changes) a revert of a change already part of the releaes requires a TSC decission/discussion.

There is some history on above PR:

- Triggered by https://github.com/nodejs/node/issues/54025 - which is fixed by this PR to my understanding. 
- There was already a fix before (https://github.com/nodejs/node/pull/54224) (marked as semver major) causeing https://github.com/nodejs/node/issues/55410 and was reverted via https://github.com/nodejs/node/pull/55414

I don't have the needed windows know how to actually fix the problem on short notice.

Which direction should we go? Revert or wait a bit for a fixup?

**Comment by Flarna** [comment]
PR to fix the issue instead of reverting: https://github.com/nodejs/node/pull/56110

**Comment by Flarna** [comment]
@nodejs/tsc Any opinion how to progress in this?
I assume the contributor docs telling that TSC decide on revert or semver major labels after the fact has a reason.

**Comment by targos** [comment]
I don't think this change is affected by https://github.com/nodejs/node/blob/main/doc/contributing/collaborator-guide.md?rgh-link-date=2024-11-30T19%3A01%3A50Z#unintended-breaking-changes.
As far as I understand, the change introduced a bug, not a breaking change in the semver sense.

**Comment by Flarna** [comment]
Well, the result of e.g. `path.resolve()` differs before and after. Even after the proposed fix it differs.

In general I'm fine with going ahead and merge the fix and let it go it's path into 23.

I assume that implies also that the `dont-land-` labels should be removed once that happens to let the initial fix float into older release lines.

**Comment by Flarna** [comment]
It's more about semver major yes/no.
if yes how to deal with it (add label/revert).
if no we can close this.

**Comment by joyeecheung** [comment]
It looks like a bug, I also agree that this should be in nodejs/node. You can send a PR to revert and maybe find a releaser who's willing to volunteer to do a patch release, unless someone stops the revert and a consensus cannot be reached, the ball isn't really in TSC's court.
